WordPress/gutenberg
# `Tabs`: Add `@radix-ui/react-tabs`-powered version of `TabPanel`

Related: #49874
Part of: #51553
This PR contains a lot of work started by @flootr (see above) with additional contributions from @ciampo and myself. While they both did much of the heavy lifting, I'm now reorganizing the effort into separate PRs.

## What?

Introduce a Radix-powered version of `TabPanel`, named `Tabs`, maintaining the same functionality and API surface as the existing component. Eventually this will be used to ensure compatibility for existing `TabPanel` implementations as we move fully to a completely new component.

## Why?

As we experiment with third-party libraries for certain UI patterns, `TabPanel` is a good candidate to explore, as there are time where additional functionality could be achieved with a more composable version of the component, which will come in future PRs

## How?

`Tabs` is a wrapper around a composed `TabPanel` utilizing the various `RadixTabs` subcomponents. `Tabs` is written to accept all the same props as `TabPanel`, passing them to the right props for the various Radix subcomponents internally.

Some key implementation bits:

**Props vs children**
The current `TabPanel` component accepts a `tabs` prop containing an object of tab details, and a `children` prop containing a function responsible for rendering the currently selected tab's content.

Radix on the other hand accepts tabs as actual children, and not a prop, so we're mapping over our `tabs` array to render the necessary `RadixTabs.Trigger` components. We do something similar for the tab contents. Here, while our `TabPanel` passes the selected tab's info to the provided `children` function, Radix expects an individual `RadixTabs.Content` component for the contents of each tab. 

**Disabled tab focus**
Our current implementation allows disabled tabs to be focused by arrow keys to make their presence known to assistive tech. This isn't something Radix supports out of the box, so we've simulated behavior triggered internally in the `Button` component by `__experimentalIsFocusable`

**asChild**
By default, Radix renders the tabs themselves as a `button` element. Using the `asChild` prop allows us to instead render them using our own `Button` component.

Note: The new component does not yet have a README, as this isn't form that `Tabs` will take long-term. I'll make sure one is added in a future PR when the new component is nearer to completion.

## Testing Instructions
1. Open Storybook and experiment with the new `Tabs` component.
2. Confirm that the controls work as expected.
3. Confirm that each story behaves the same as its `TabPanel` counterpart.
4. Specifically test keyboard interactions. Disabled tabs should be selectable via arrow keys, but not pointer events.
